tools/cloud-build/provision/list_tests.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import glob
+import json
+
+"""
+List build files in `tools/cloud-build/daily-tests/builds/`
+and creates a cron-schedule-string for each of them.
+E.g.:
+```
+$ ./list_tests.py | jq
+{
+  "batch-mpi": "30 0 * * *",
+  "chrome-remote-desktop": "48 0 * * *",
+  "cloud-batch": "6 1 * * *",
+  "hpc-high-io-v4": "24 1 * * *",
+  "hpc-high-io-v5": "42 1 * * *",
+...
+```
+"""
+def list_builds():
+    builds = glob.glob("*.yaml", root_dir="../daily-tests/builds/")
+    assert builds, "No builds have been found"
+    builds = sorted([b[:-5] for b in builds])
```

Comment:
I personally found it difficult to understand line 37 as it was not immediately clear that you were not only resorting the builds, but also removing the .yaml from the end. It may be worth adding a comment before line 37.

Reply:
Addressed